packages/components/src/custom-select-control-v2/styles.ts

```diff
@@ -96,6 +100,11 @@ export const SelectPopover = styled( Ariakit.SelectPopover )`
 	border-radius: 2px;
 	background: ${ COLORS.theme.background };
 	border: 1px solid ${ COLORS.theme.foreground };
+	z-index: 9999; // Ensure the popover is on top
+	position: absolute !important;
+	max-height: 400px;
+	overflow: auto;
+	min-width: 100%;
```

Comment:
| Before | After |
|---------|---------|
|  <img width="1220" alt="Screenshot 2024-06-07 at 12 56 33 p m" src="https://github.com/WordPress/gutenberg/assets/81248/0e97f6fc-70ac-45b7-ad61-e827713aee96"> | <img width="1224" alt="Screenshot 2024-06-07 at 12 55 16 p m" src="https://github.com/WordPress/gutenberg/assets/81248/2b621b39-8fc9-42c4-a694-c1816391c6c7">
 |

Reply:
I can't replicate the popover width issue on Storybook — plus, it seems like [we're already passing the `sameWidth` prop to the underlying Ariakit implementation ](https://github.com/WordPress/gutenberg/blob/6fa347111e9714141c0a38711b518d9ef2485485/packages/components/src/custom-select-control-v2/custom-select.tsx#L113)